src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1291,38 +1196,44 @@ public static string GetModulePath(string currentProcessModulePath, string hklmM
                 {
                     currentProcessModulePath += hklmMachineModulePath; // += EVT.Machine
                 }
+
+#if !UNIX
+                // Add Windows Modules path
+                string windowsPowerShellModulePath = GetExpandedEnvironmentVariable("SystemDrive", EnvironmentVariableTarget.Process) + @"\WINDOWS\system32\WindowsPowerShell\v1.0\Modules";
+                currentProcessModulePath = $"{currentProcessModulePath}{Path.PathSeparator}{windowsPowerShellModulePath}";
+#endif
             }
             // EVT.Process exists
             // Now handle the case where the environment variable is already set.
             else
             {
-                // When SxS PS instance A starts SxS PS instance B, A's PSHome module path might be inherited by B. We need to remove that path from B
-                currentProcessModulePath = RemoveSxSPsHomeModulePath(currentProcessModulePath, personalModulePath, sharedModulePath, psHomeModulePath);
-
                 string personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? personalModulePath : hkcuUserModulePath;
                 string systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? psHomeModulePath : hklmMachineModulePath;
 
+                // Maintain order of the paths, but ahead of any existing paths:
+                // personalModulePath
+                // sharedModulePath
+                // systemModulePath
                 currentProcessModulePath = AddToPath(currentProcessModulePath, personalModulePathToUse, 0);
 
                 int insertIndex = -1;
-#if !UNIX
-                string windowsPowerShellModulePath = GetWindowsPowerShellPSHomeModulePath();
-                // If the Windows PowerShell Module path is already present, insert the system module path
-                // ($PSHOME/Modules) before it.
-                insertIndex = PathContainsSubstring(currentProcessModulePath, windowsPowerShellModulePath);
-#endif
-                currentProcessModulePath = AddToPath(currentProcessModulePath, systemModulePathToUse, insertIndex);
-            }
 
-            // if we reached this point - always add <Program Files> location to EVT.Process
-            // everything below is the same behaviour as WMF 4 code
+                insertIndex = PathContainsSubstring(currentProcessModulePath, personalModulePath);
```

Comment:
I'm a little confused here, why searching for `personalModulePath` instead of `personalModulePathToUse`?
Note that you are using `systemModulePathToUse` below when calling `AddToPath`.

Reply:
That was just a mistake, will update.